Allow files without .txt extension as argument to --list

#### What is the purpose of your *issue*?
- [ ] Bug
- [x ] Feature Request
- [ ] Question
- [ ] Other

### Description
Some files failed to download, I wanted to retry them and copied the url's to a file called "tst"
When I pass that file to spotdl it says
error: tst is not of a valid argument to --list, argument must be plain text file.

If I rename the file to tst.txt the file is accepted.
On linux it is not uncommon to have ascii files without extension.

Suggestion is to remove the restriction (and maybe use a different way to determine if the file is ascii)

I've submitted a small pull request to check if the file passed with --list has valid content.
Basically this commit only accepts text files for which each line starts with http (and that includes https)
Of course the test can be changed to allow e.g. empty lines, whitespace, comments, ... but I decided to only do this, just to test the waters.

Note that this is my first pull request to this repo, so apologies upfront if I made a mistake left or right.
Also my python is a bit rusty.

Code has been tested locally.

I also tried to run pytest, but somehow all tests fails with messages like this one:
_______________________________________________________________________ ERROR collecting spotdl/encode/encoders/tests/test_ffmpeg.py _______________________________________________________________________
ImportError while importing test module '/home/fmeulenbroeks/git/spotify-downloader/spotdl/encode/encoders/tests/test_ffmpeg.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
E   ModuleNotFoundError: No module named 'spotdl.encode.encoders.tests'

I have no idea why this is, this is not even near to or related to what I changed.
(my development system is linux if that matters, and yes, I recently used pytest in another project without problems)

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -8,6 +8,8 @@ The release dates mentioned follow the format `DD-MM-YYYY`.
 
 ## [Unreleased]
 
+- Improve input file validation for --list (#818)
+
 ## [2.2.2] - 28-07-2020
 ## Fixed
 - PyTube failing on outdated regex. (#798)

--- a/spotdl/command_line/arguments.py
+++ b/spotdl/command_line/arguments.py
@@ -1,7 +1,6 @@
 import appdirs
 
 import argparse
-import mimetypes
 import os
 import sys
 import shutil
@@ -323,14 +322,16 @@ class ArgumentHandler:
     def run_errands(self):
         args = self.configured_args
 
-        if (args.get("list")
-            and not mimetypes.MimeTypes().guess_type(args["list"])[0] == "text/plain"
-        ):
-            raise ArgumentError(
-                "{0} is not of a valid argument to --list, argument must be plain text file.".format(
-                    args["list"]
-                )
-            )
+        if (args.get("list")):
+            textchars = bytearray({7,8,9,10,12,13,27} | set(range(0x20, 0x7e)))
+            is_binary_string = lambda bytes: bool(bytes.translate(None, textchars))
+            with open(args["list"], 'rb') as file:
+                if is_binary_string(file.read(1024)):
+                    raise ArgumentError(
+                        "{0} is not of a valid argument to --list, it must be a text file".format(
+                            args["list"]
+                        )
+                    )
 
         if args.get("write_m3u") and not args.get("list"):
             raise ArgumentError("--write-m3u can only be used with --list.")

